panacea/web/static/js/app.js

```diff
@@ -1,104 +1,38 @@
 import 'babel-polyfill';
 import 'bootstrap';
 import 'whatwg-fetch';
-import * as Request from './request';
+import { RequestChain } from './request-chain';
 import * as View from './view';
 
-document.getElementById('file-form').addEventListener('submit', e => {
+View.fileForm.addEventListener('submit', e => {
   e.preventDefault();
-  submitFile.bind(e.target)();
+  const formData = new FormData(e.target);
+  submitFile(formData);
+  e.target.reset();
 });
 
-async function submitFile() {
+const submitFile = formData => {
   View.hideFileForm();
   View.hideResults();
   try {
-    await handleDrugsResponse(await Request.drugs(new FormData(this)));
+    retrievePML(pml => {
+      const requestChain = new RequestChain(pml);
+      requestChain.start(formData);
```

Comment:
We **do** need await, I didn't notice but our loading icon wasn't showing up at all because of synchronicity issues. Great spot, I have just pushed a fix 🥇